app/views/courses/TeacherClassesView.coffee

```diff
@@ -0,0 +1,127 @@
+RootView = require 'views/core/RootView'
+template = require 'templates/courses/teacher-classes-view'
+Classroom = require 'models/Classroom'
+Classrooms = require 'collections/Classrooms'
+Courses = require 'collections/Courses'
+Campaign = require 'models/Campaign'
+Campaigns = require 'collections/Campaigns'
+LevelSessions = require 'collections/LevelSessions'
+CourseInstance = require 'models/CourseInstance'
+CourseInstances = require 'collections/CourseInstances'
+ClassroomSettingsModal = require 'views/courses/ClassroomSettingsModal'
+InviteToClassroomModal = require 'views/courses/InviteToClassroomModal'
+User = require 'models/User'
+utils = require 'core/utils'
+helper = require 'lib/coursesHelper'
+
+module.exports = class TeacherClassesView extends RootView
+  id: 'teacher-classes-view'
+  template: template
+  
+  events:
+    'click .edit-classroom': 'onClickEditClassroom'
+    'click .archive-classroom': 'onClickArchiveClassroom'
+    'click .unarchive-classroom': 'onClickUnarchiveClassroom'
+    'click .add-students-button': 'onClickAddStudents'
+    'click .create-classroom-button': 'onClickCreateClassroom'
+
+  constructor: (options) ->
+    super(options)
+    @classrooms = new Classrooms()
+    @classrooms.fetchMine()
+    @supermodel.trackCollection(@classrooms)
+    @listenTo @classrooms, 'sync', ->
+      for classroom in @classrooms.models
+        classroom.sessions = new LevelSessions()
+        jqxhrs = classroom.sessions.fetchForAllClassroomMembers(classroom)
+        if jqxhrs.length > 0
+          @supermodel.trackCollection(classroom.sessions)
+    
+    @courses = new Courses()
+    @courses.fetch()
+    @supermodel.trackCollection(@courses)
+    
+    @campaigns = new Campaigns()
+    @campaigns.fetchByType('course')
+    @supermodel.trackCollection(@campaigns)
+    
+    @courseInstances = new CourseInstances()
+    @courseInstances.fetchByOwner(me.id)
+    @supermodel.trackCollection(@courseInstances)
+    @progressDotTemplate = require 'templates/courses/progress-dot'
+    
+    # Level Sessions loaded after onLoaded to prevent race condition in calculateDots
+  
+  afterRender: ->
+    super()
+    $('.progress-dot').each (i, el) ->
+      dot = $(el)
+      dot.tooltip({
+        html: true
+        container: dot
+        template: '<div class="tooltip" role="tooltip"><div class="tooltip-arrow"></div><div class="tooltip-arrow-cover"></div><div class="tooltip-inner"></div></div>'
+      })
```

Comment:
Update to use the new tooltip styles. A special template is no longer required, everything is done with css to any tooltip within a `.style-flat`.